homeassistant/components/utility_meter/services.yaml

```diff
@@ -1,35 +1,41 @@
 # Describes the format for available switch services
 
 reset:
+  name: Reset
   description: Resets the counter of an utility meter.
-  fields:
-    entity_id:
-      description: Name(s) of the utility meter to reset
-      example: "utility_meter.energy"
+  target:
 
 next_tariff:
+  name: Next Tariff
   description: Changes the tariff to the next one.
-  fields:
-    entity_id:
-      description: Name(s) of entities to reset
-      example: "utility_meter.energy"
+  target:
 
 select_tariff:
-  description: selects the current tariff of an utility meter.
+  name: Select Tariff
+  description: Selects the current tariff of an utility meter.
+  target:
   fields:
-    entity_id:
-      description: Name of the entity to set the tariff for
-      example: "utility_meter.energy"
     tariff:
+      name: Tariff
       description: Name of the tariff to switch to
       example: "offpeak"
+      required: true
+      selector:
+        text:
 
 calibrate:
-  description: calibrates an utility meter.
+  name: Calibrate
+  description: Calibrates a utility meter sensor.
+  target:
+    entity:
+      domain: sensor
+      integration: utility_meter
```

Comment:
```suggestion
```

Reply:
sure I can delete, but it will only show the utility_meter.* entities.... not the sensor entities created by utility_meter....

Reply:
Yes, so you want both `sensor` and `utility_meter` entities? Because that is not possible, then you should not filter at all:

```
target:
  entity:
```

Reply:
no, I just want the `sensor` entities created by the `utility_meter` integration

Reply:
OK, but `utility_meter` doesn't have a config entry?

So then you can't filter by integration. 

```
target:
  entity:
    domain: sensor
```

Reply:
OK, so that is what I'm missing I suppose, in order for the filtering to work utility_meter must have a config_entry ?

I never implemented config_entry for utility_meter because it's internal and was told that config_entry was reserved for external integration, but if that's the way forward I will implement it

Reply:
You should not. But the entities are not even in the entity registry, we don't have any information we could filter those on 🤷‍♂️

Reply:
OK... so this is what's currently possible

PR is ready then